skins/oasis/modules/ContentReviewModuleController.class.php

```diff
@@ -12,10 +20,118 @@ class ContentReviewModuleController extends WikiaController {
+			$latestStatus = $this->prepareTemplateData( self::STATUS_LIVE, $latestRevisionId );
+		} elseif ( $latestRevisionId === (int) $pageStatus['latestId']
+			&& Helper::isStatusAwaiting( $pageStatus['latestStatus'] ) )
+		{
+			$latestStatus = $this->prepareTemplateData( self::STATUS_AWAITING, $latestRevisionId );
+		} elseif ( $latestRevisionId === (int) $pageStatus['lastReviewedId']
+			&& (int) $pageStatus['lastReviewedStatus'] === ReviewModel::CONTENT_REVIEW_STATUS_REJECTED )
+		{
+			$latestStatus = $this->prepareTemplateData( self::STATUS_REJECTED, $latestRevisionId, true );
+		} elseif ( $latestRevisionId > (int) $pageStatus['liveId']
+			&& $latestRevisionId > (int) $pageStatus['latestId']
+			&& $latestRevisionId > (int) $pageStatus['lastReviewedId'] ) {
+			$latestStatus = $this->prepareTemplateData( self::STATUS_UNSUBMITTED, $latestRevisionId );
+		}
+
+		return $latestStatus;
+	}
+
+	public function getLastRevisionStatus( array $pageStatus ) {
+		if ( is_null( $pageStatus['lastReviewedId'] ) ) {
+			$lastStatus = $this->prepareTemplateData( self::STATUS_NONE );
+		} elseif ( (int) $pageStatus['lastReviewedStatus'] === ReviewModel::CONTENT_REVIEW_STATUS_APPROVED ) {
+			$lastStatus = $this->prepareTemplateData( self::STATUS_APPROVED, $pageStatus['lastReviewedId'] );
+		} elseif ( (int) $pageStatus['lastReviewedStatus'] === ReviewModel::CONTENT_REVIEW_STATUS_REJECTED ) {
+			$lastStatus = $this->prepareTemplateData( self::STATUS_REJECTED, $pageStatus['lastReviewedId'], true );
+		}
+
+		return $lastStatus;
+	}
+
+	public function getLiveRevisionStatus( array $pageStatus ) {
+		if ( !is_null( $pageStatus['liveId'] ) ) {
+			$liveStatus = $this->prepareTemplateData( self::STATUS_LIVE, $pageStatus['liveId'] );
+		} else {
+			$liveStatus = $this->prepareTemplateData( self::STATUS_NONE );
+		}
+
+		return $liveStatus;
+	}
+
+	public function prepareTemplateData( $status, $revisionId = null, $withReason = false ) {
+		$templateData = [
+			'statusKey' => $status,
+			'message' => wfMessage( "content-review-module-status-{$status}" )->escaped(),
+		];
+
+		if ( !is_null( $revisionId ) ) {
+			$templateData['diffLink'] = $this->createRevisionLink( $revisionId );
+			if ( $withReason ) {
+				$templateData['reasonLink'] = $this->createRevisionTalkpageLink( $revisionId );
+			}
+		}
+
+		return $templateData;
+	}
+
+	public function createRevisionLink( $revisionId ) {
+		return Linker::linkKnown(
+			$this->getContext()->getTitle(),
+			"#{$revisionId}",
+			[
+				'target' => '_blank'
+			],
+			[
+				'diff' => $revisionId,
+			]
+		);
+	}
 
-		$this->moduleType = $params['moduleType'];
-		$this->isTestModeEnabled = Wikia\ContentReview\Helper::isContentReviewTestModeEnabled();
+	public function createRevisionTalkpageLink( $revisionId ) {
+		return Linker::linkKnown(
+			$this->getContext()->getTitle()->getTalkPage(),
+			wfMessage( 'content-review-rejection-reason-link' )->escaped(),
+			[
+				'target' => '_blank'
```

Comment:
No reason to decide for user to open link in new tab. Especially that opening in same window doesn't break anything.